cmd/new.go

```diff
@@ -1,6 +1,7 @@
 package cmd
```

Comment:
The code in this PR doesn't seem wrong, I will have to try it out myself to be sure. But it is quite a bit of changes and I just want to give it a try myself before merging. I will try and get to it this weekend, I may create my own PR or edit this one, not sure yet.

One thing is your change from change to prompt.go is a fairly large change and I want to make sure I fully understand it. Thanks for the work though this is a big help for sure.

Reply:
Yep, besides being a big change, I don't think that the "prompt change" is the correct abstraction, btw it's hard to find the proper naming for the layer between the `new` command and the atomic change itself.

Personally I would try to avoid moving all of it to the `cmd` package.

Definitely, feel free to update this PR or open a new one, I'm still here if any help is needed!

Thank you!!